api/valor_api/backend/core/dataset.py

```diff
@@ -254,6 +254,15 @@ def set_dataset_status(
             dataset_names=[name],
         ):
             raise exceptions.EvaluationRunningError(dataset_name=name)
+    elif status == enums.TableStatus.FINALIZED:
+        datum_count = (
+            db.query(func.count(models.Datum.id))
```

Comment:
nit: this logic might be better encapsulated in a small, seperate function (e.g., `_validate_dataset_contains_datums`)